modules/SofaGraphComponent/src/SofaGraphComponent/SceneCheckCollisionResponse.cpp

```diff
@@ -0,0 +1,113 @@
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include "SceneCheckCollisionResponse.h"
+
+#include <sofa/simulation/Node.h>
+#include <SofaBaseCollision/DefaultContactManager.h>
+#include <sofa/core/behavior/BaseAnimationLoop.h>
+#include <sofa/core/behavior/ConstraintSolver.h>
+
+namespace sofa::simulation::_scenechecking_
+{
+
+using sofa::simulation::Node;
+
+const std::string SceneCheckCollisionResponse::getName()
+{
+    return "SceneCheckCollisionResponse";
+}
+
+const std::string SceneCheckCollisionResponse::getDesc()
+{
+    return "Check there is not duplicated name in the scenegraph";
+}
+
+void SceneCheckCollisionResponse::doInit(Node* node)
+{
+    SOFA_UNUSED(node);
+    m_message.str("");
+}
+
+void SceneCheckCollisionResponse::doCheckOn(Node* node)
+{
+    if(m_checkDone)
+        return;
+
+    const sofa::core::objectmodel::BaseContext* root = node->getContext()->getRootContext();
+    std::vector<sofa::component::collision::DefaultContactManager*> contactManager;
+    root->get<sofa::component::collision::DefaultContactManager>(&contactManager, core::objectmodel::BaseContext::SearchDown);
+    Size nbContactManager = contactManager.size();
+    if( nbContactManager == 0 )
+    {
+        m_checkDone=true;
+    }
+    else
+    {
+        m_checkDone=true;
+        if( nbContactManager!= 1 )
+        {
+            m_message << "Only one DefaultContactManager is needed."<< msgendl;
+        }
+        else
+        {
+            std::string response = contactManager[0]->response.getValue().getSelectedItem();
+
+            if ( response == "StickContactConstraint" )
+            {
+                sofa::core::behavior::BaseAnimationLoop* animationLoop;
+                root->get(animationLoop, sofa::core::objectmodel::BaseContext::SearchRoot);
+                if (!animationLoop || ( animationLoop && ( animationLoop->getClassName() != "FreeMotionAnimationLoop" )) )
+                {
+                    m_message <<"A FreeMotionAnimationLoop must be in the scene to solve StickContactConstraint" << msgendl;
+                }
```

Comment:
What if there is an animation loop inheriting from `FreeMotionAnimationLoop`? Maybe you could use a `dynamic_cast` or `BaseClass::hasParent`.

Reply:
The comment is also valid for your other searches

Reply:
I understand. Maybe (in a future PR) we can think about adding a SceneChecker from any module/plugin. If we were able to do that, we could put the scene checker in SofaConstraint